darwin/client.py

```diff
@@ -189,7 +189,7 @@ def get_remote_dataset(
             parsed_dataset_identifier.team_slug = self.default_team
 
         try:
-            matching_datasets: List[RemoteDataset] = [
+            matching_datasets: List[RemoteDatasetV2] = [
```

Comment:
These type changes aren't functionally significant, but since we no longer have RemoteDatasetV1 they make some typechecker errors less likely

Reply:
There are still some places (docstrings/types) where we use `RemoteDataset` - should we change all of them 🤔